### DISCUSSION TITLE
Is it possible to use java-snapshot-testing with Cucumber (features and steps) and Junit4?

### DISCUSSION THREAD

**Discussion by maastha** [origin-energy/java-snapshot-testing#168] [General]
Since Cucumber has its own test runner; annotating on the step class won't extend or change the cucumber test runner to be able to inject `Expect expect` instance variable.

Could you please share guidance/example of how to use this library in Cucumber steps?
`Expect expect = Expect.of()` requires a `java.lang.reflect.Method method` which I don't think is accessible in Cucumber steps.

**Comment by jackmatt2** [comment]
I have never tried this library with Cucumber.  

Cucumber uses shared steps with actual tests being written in Gherkin syntax.  As there is no uniqueness with the Java Test method it won't be able to identify the comparator test correctly. You would be forced using @SnapshotName or an equivalent to uniquely identity each test - this may require some small modifications to java-snapshot-testing.

So unfortunately, unless you implement Gherkin integration and annotation it probably won't work.

You might be able to hooking the "Scenario" keyword for the snapshot name

**Reply by maastha** [reply]
>You might be able to hooking the "Scenario" keyword for the snapshot name

Thanks a lot @jackmatt2 for your response, really appreciate it!

I think it would be possible to access data regarding the feature and the scenario in question. 

My understanding is it ultimately boils down to the name of the Snapshot, is that correct? 
If yes, then that means we would need a way to get/set the Snapshot name at runtime based on certain values (for example-feature name, scenario name, input, tags etc) or really any custom name for the snapshot instead of only what can be set via @SnapshotName. 
What are your thoughts on that? 
In this case I'm not sure I understand how an annotation would work since it'll need to be set at runtime.

**Reply by jackmatt2** [reply]
Yes that's correct. You just need some way to uniquely identify the test without any risk of collisions. When the comparison takes place - it needs to be able to identity which snapshot to compare to.
